src/optimise/train.jl

```diff
@@ -80,6 +80,33 @@ end
 batchmemaybe(x) = tuple(x)
 batchmemaybe(x::Tuple) = x
 
+"""
+    step!(loss, params, opt)
+
+`step!` uses a `loss` function (with no inputs) to improve the [Model parameters](@ref) (`params`)
+based on a pluggable [Optimisers](@ref) (`opt`). It represents a single step in
+the training loop `train!`. While there is a default implementation for
+optimisers which are based on the `update!` function and only require gradient
+information, this `step!` has to be overloaded for more general optimisers.
+
+While the loss function of `train!` still accepts data as input, the loss function
+of `step!` accepts no input. `train!` cycles through the data in a loop
+roughly like this
+
+```julia
+for d in data
+  step!(ps, opt) do
+    loss(d)
+  end
+```
+
```

Comment:
````suggestion
Unlike `train!`, the loss function of `step!` accepts no input.
Instead, `train!` cycles through the data in a loop and calls `step!`:
```julia
for d in data
  step!(ps, opt) do
    loss(d)
  end
end
```
If you are writing [Custom Training loops](@ref), then you should follow this pattern.
````